### PR TITLE
Switched from raven to sentry sdk

### PR DESCRIPTION
**Pull request by felixoi**
As mentioned in #270 raven is deprecated.

### CODE REVIEW

File: spongeauth/spongeauth/settings/prod.py [hunk 1]
```diff
@@ -63,10 +59,11 @@
     d = SSO_ENDPOINTS.setdefault(name, {})
     d[key.lower()] = v
 
-RAVEN_CONFIG = {
-    'dsn': os.environ['RAVEN_DSN'],
-    'release': raven.fetch_git_sha(GIT_REPO_ROOT),
-}
+sentry_sdk.init(
+    dsn=os.environ['RAVEN_DSN'],
```
Comment (felixoi) [inline spongeauth/spongeauth/settings/prod.py]: I could rename this one to `SENTRY_DSN`. Would need a change in the env settings of the production machine.

Comment (lukegb) [reply spongeauth/spongeauth/settings/prod.py]: Slight preference for 

```suggestion
    dsn=os.environ.get('RAVEN_DSN'),
```

Specifically, it's useful to have usernames/user IDs when debugging issues with auth, and also if you use `.get` then we can easily swap RAVEN_DSN to SENTRY_DSN.

File: spongeauth/spongeauth/settings/utils.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import os.path
+
+def fetch_git_sha(path, head=None):
```
Comment (lukegb) [inline spongeauth/spongeauth/settings/utils.py:29]: This code looks like it's copied from Raven, and is at least missing its copyright header. Hmm.

File: spongeauth/spongeauth/settings/prod.py [hunk 3]
```diff
@@ -63,10 +59,11 @@
...
+    integrations=[DjangoIntegration()],
+    release=fetch_git_sha(GIT_REPO_ROOT)
```
Comment (lukegb) [inline spongeauth/spongeauth/settings/prod.py]:
```suggestion
    release=fetch_git_sha(GIT_REPO_ROOT),
    send_default_pii=True,
```

Comment (lukegb) [reply spongeauth/spongeauth/settings/prod.py]: (please leave the trailing comma, basically)

File: spongeauth/spongeauth/settings/prod.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
-import os
-import os.path
-import raven
+import sentry_sdk
+from sentry_sdk.integrations.django import DjangoIntegration
 
+from spongeauth.spongeauth.settings.utils import fetch_git_sha
```
Comment (lukegb) [inline spongeauth/spongeauth/settings/prod.py]:
```suggestion
from .utils import fetch_git_sha
```

**Review by lukegb — CHANGES_REQUESTED** [review]
Thanks for the contribution!

File: spongeauth/spongeauth/settings/utils.py (hunk 2 above)
Comment (felixoi) [reply spongeauth/spongeauth/settings/utils.py:29]: Yeah right, missing it. I was unsure if you need it or not, I asked the sentry dev for the method in the new sdk and he said there is no one. So the suggested to just copy it. Any other suggestions?

Edit: Added the copyright header.

Comment (lukegb) [reply spongeauth/spongeauth/settings/utils.py:29]: Sounds good, thanks.

**Review by lukegb — APPROVED** [review]

**Pull request merged**